src/custom/pages/Profile/index.tsx

```diff
@@ -27,14 +27,16 @@ import { MouseoverTooltipContent } from 'components/Tooltip'
 import NotificationBanner from 'components/NotificationBanner'
 import { SupportedChainId as ChainId } from 'constants/chains'
 import AffiliateStatusCheck from 'components/AffiliateStatusCheck'
+import { useGpOrders } from 'api/gnosisProtocol/hooks'
 
 export default function Profile() {
   const referralLink = useReferralLink()
   const { account, chainId } = useActiveWeb3React()
   const { profileData, isLoading, error } = useFetchProfile()
   const lastUpdated = useTimeAgo(profileData?.lastUpdated)
   const isTradesTooltipVisible = account && chainId == 1 && !!profileData?.totalTrades
-  const userHasTrades = !!profileData?.totalTrades
+  const gpOrders = useGpOrders(account)
+  const hasOrders = (gpOrders?.length || 0) > 0
```

Comment:
Refactoring suggestion: given that you are doing this twice, you can create a hook for this similar to `hasTrades`

Reply:
The problem is `useGpOrders` is already a hook so moving it to a function similar to `hasTrades` would not be possible. I could instead fetch orders api inside a function call `hasOrders` but I would be repeating the same logic inside de hook.

Reply:
What about a hook called `useHasTrades(account:string) => boolean | undefined` ?

No big deal, I'm ok if you don't agree